### PR TITLE
Unify unattended single artifact

### PR DESCRIPTION
**Pull request by verdx**
|Related issue|
|---|
| #1232  |

## Description

This PR creates a new script, `builder.sh`, which builds a file where all functions and file templates needed for the unattended installation are. This new file serves as a single installation script, with no need to download resources from anywhere. 
The script, `builder.sh`, also creates `wazuh-passwords-tool.sh` and `wazuh-cert-tool.sh`

## Tests

- Build the package in any supported platform
  - [x] All-in-one CentOS 7
  - [x] Distributed Mixed on CentOS 7, Debian 9 and Ubuntu 18

### CODE REVIEW

File: unattended_installer/builder.sh [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+# Tool to create wazuh_install.sh, wazuh-cert-tool.sh
+# and wazuh-passwords-tool.sh
+# Copyright (C) 2015, Wazuh Inc.
+#
+# This program is a free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+base_path="/home/verdx/Documents/Wazuh/entorno/wazuh-packages/unattended_installer"
```
Comment (DFolchA) [inline unattended_installer/builder.sh:12]: Please change base path

**Review by DFolchA — CHANGES_REQUESTED** [review]

File: unattended_installer/builder.sh (hunk 1 above)
Comment (verdx) [reply unattended_installer/builder.sh]: Changed in eb83cc7ec604c15342e4fe987303b0347a6d0d9b

**Comment by DFolchA** [comment]
# Use case study

## Use case 1: Build installer script

```
root@ubuntu:/home/vagrant/repo# bash builder.sh -i -d
root@ubuntu:/home/vagrant/repo# ls -la wazuh-install.sh 
-rwx--x--x 1 vagrant vagrant 109935 Feb 24 15:43 wazuh-install.sh

```



## Use case 1: See help

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -h

NAME
        wazuh-install.sh - Install and configure Wazuh central components.

SYNOPSIS
        wazuh-install.sh [OPTIONS] -a | -c | -s | -wi <indexer-node-name> | -wd <dashboards-node-name> | -ws <wazuh-node-name>

DESCRIPTION
        -a,  --all-in-one
                Install and configure Wazuh server, Wazuh indexer, Wazuh dashboard and filebeat

        -c,  --configfile <path-to-config-yml>
                Path to config file. By default: /home/vagrant/repo/config.yml

        -ds,  --disable-spinner
                Disables the spinner indicator.

        -F,  --force-dashboard
                Ignore indexer cluster related errors in Wazuh Dashboard installation

        -g,  --generate-configurations
                Generate /home/vagrant/repo/wazuh-install-files.tar file from /home/vagrant/repo/config.yml containing the files that will be needed for installation. You will need to copy this file to other hosts in distributed deployments.

        -h,  --help
                Display this help and exit.

        -i,  --ignore-hardware-check
                Ignore check for minimum hardware requirements

        -o,  --overwrite
                Overwrites previously installed components. This will erase all the existing configuration and data.

        -s,  --start-cluster
                Starts the indexer cluster.

        -t,  --tar <path-to-certs-tar>
                Path to tar containing certificate files. By default: /home/vagrant/repo/wazuh-install-files.tar

        -u,  --uninstall
                Uninstalls all Wazuh components. This will erase all the existing configuration and data.

        -v,  --verbose
                Shows the complete installation output.

        -V,  --version
                Shows the version of the script and installed packages

        -wd,  --wazuh-dashboards <dashboards-node-name>
                Install and configure Wazuh dashboard.

        -wi,  --wazuh-indexer <indexer-node-name>
                Install and configure Wazuh indexer.

        -ws,  --wazuh-server <wazuh-node-name>
                Install and configure Wazuh server and filebeat.

```



## Use case 2: Install All In One

### 2.1: Missing dependencies

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a
24/02/2022 16:10:17 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 16:10:19 INFO: --- Configuration files ---
24/02/2022 16:10:19 INFO: Generating configuration files.
24/02/2022 16:10:19 INFO: Created /home/vagrant/repo/wazuh-install-files.tar. Contains Wazuh cluster key, certificates, and passwords necessary for installation.
24/02/2022 16:10:26 INFO: --- Dependencies ----
24/02/2022 16:10:26 INFO: Installing apt-transport-https.
24/02/2022 16:10:27 INFO: Installing unzip.
24/02/2022 16:10:33 INFO: --- Wazuh indexer ---
24/02/2022 16:10:33 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 16:11:41 INFO: Wazuh indexer installation finished.
24/02/2022 16:11:41 INFO: Wazuh indexer post-install configuration finished.
24/02/2022 16:11:41 INFO: Starting service wazuh-indexer.
24/02/2022 16:11:49 INFO: Wazuh-indexer service started.
24/02/2022 16:11:49 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 16:11:54 INFO: Wazuh indexer cluster started.
24/02/2022 16:11:54 INFO: --- Wazuh server ---
24/02/2022 16:11:54 INFO: Starting the Wazuh manager installation.
24/02/2022 16:12:37 INFO: Wazuh manager installation finished.
24/02/2022 16:12:37 INFO: Starting service wazuh-manager.
24/02/2022 16:12:54 INFO: Wazuh-manager service started.
24/02/2022 16:12:54 INFO: Starting Filebeat installation.
24/02/2022 16:13:00 INFO: Filebeat installation finished.
24/02/2022 16:13:01 INFO: Filebeat post-install configuration finished.
24/02/2022 16:13:01 INFO: Starting service filebeat.
24/02/2022 16:13:02 INFO: Filebeat service started.
24/02/2022 16:13:02 INFO: --- Wazuh dashboard ---
24/02/2022 16:13:02 INFO: Starting Wazuh dashboard installation.
24/02/2022 16:13:47 INFO: Wazuh dashboard installation finished.
24/02/2022 16:13:47 INFO: Wazuh dashboard post-install configuration finished.
24/02/2022 16:13:47 INFO: Starting service wazuh-dashboard.
24/02/2022 16:13:47 INFO: Wazuh-dashboard service started.
24/02/2022 16:14:00 INFO: Starting Wazuh dashboard.
24/02/2022 16:14:11 INFO: Wazuh dashboard started.
24/02/2022 16:14:11 INFO: --- Summary ---
24/02/2022 16:14:11 INFO: You can access the web interface https://<wazuh-dashboard-host-ip>.
    User: admin
    Password: kbd3sEOdkvtHnLDsz6h4Wh2s2xxr6k3L
24/02/2022 16:14:11 INFO: Installation finished.
24/02/2022 16:14:11 INFO: The certificates and passwords used are stored in /home/vagrant/repo/wazuh-install-files.tar.

```



### 2.2: All dependencies already installed

```
24/02/2022 16:17:50 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 16:18:02 INFO: --- Configuration files ---
24/02/2022 16:18:02 INFO: Generating configuration files.
24/02/2022 16:18:03 INFO: Created /home/vagrant/repo/wazuh-install-files.tar. Contains Wazuh cluster key, certificates, and passwords necessary for installation.
24/02/2022 16:18:10 INFO: --- Wazuh indexer ---
24/02/2022 16:18:10 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 16:19:19 INFO: Wazuh indexer installation finished.
24/02/2022 16:19:19 INFO: Wazuh indexer post-install configuration finished.
24/02/2022 16:19:19 INFO: Starting service wazuh-indexer.
24/02/2022 16:19:27 INFO: Wazuh-indexer service started.
24/02/2022 16:19:27 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 16:19:32 INFO: Wazuh indexer cluster started.
24/02/2022 16:19:32 INFO: --- Wazuh server ---
24/02/2022 16:19:32 INFO: Starting the Wazuh manager installation.
24/02/2022 16:20:00 INFO: Wazuh manager installation finished.
24/02/2022 16:20:00 INFO: Starting service wazuh-manager.
24/02/2022 16:20:15 INFO: Wazuh-manager service started.
24/02/2022 16:20:15 INFO: Starting Filebeat installation.
24/02/2022 16:20:20 INFO: Filebeat installation finished.
24/02/2022 16:20:22 INFO: Filebeat post-install configuration finished.
24/02/2022 16:20:22 INFO: Starting service filebeat.
24/02/2022 16:20:23 INFO: Filebeat service started.
24/02/2022 16:20:23 INFO: --- Wazuh dashboard ---
24/02/2022 16:20:23 INFO: Starting Wazuh dashboard installation.
24/02/2022 16:21:07 INFO: Wazuh dashboard installation finished.
24/02/2022 16:21:07 INFO: Wazuh dashboard post-install configuration finished.
24/02/2022 16:21:07 INFO: Starting service wazuh-dashboard.
24/02/2022 16:21:07 INFO: Wazuh-dashboard service started.
24/02/2022 16:21:20 INFO: Starting Wazuh dashboard.
24/02/2022 16:21:32 INFO: Wazuh dashboard started.
24/02/2022 16:21:32 INFO: --- Summary ---
24/02/2022 16:21:32 INFO: You can access the web interface https://<wazuh-dashboard-host-ip>.
    User: admin
    Password: JhlkyF4oKjijUsTJt6b0etEC2qYVmL19
24/02/2022 16:21:32 INFO: Installation finished.
24/02/2022 16:21:32 INFO: The certificates and passwords used are stored in /home/vagrant/repo/wazuh-install-files.tar

```

## 2.3: Installation already in place

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a
24/02/2022 16:26:06 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 16:26:07 ERROR: Wazuh manager already installed.
24/02/2022 16:26:07 ERROR: Wazuh indexer already installed.
24/02/2022 16:26:07 ERROR: Wazuh dashboard already installed.
24/02/2022 16:26:07 ERROR: Filebeat already installed.
24/02/2022 16:26:07 INFO: If you want to overwrite the current installation, run this script adding the option -o/--overwrite. This will erase all the existing configuration and data.
```

## 2.4: -o option alone

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -o
24/02/2022 16:33:59 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 16:34:00 ERROR: The argument -o|--overwrite must be used in conjunction with -a|--all-in-one, -wd|--wazuh-dashboard, -wi|--wazuh-indexer, or -ws|--wazuh-server.
```

### 2.5: Overwrite installation

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -o
24/02/2022 16:35:08 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 16:35:09 INFO: --- Removing existing Wazuh installation ---
24/02/2022 16:35:09 INFO: Removing Wazuh manager.
24/02/2022 16:35:15 INFO: Wazuh manager removed.
24/02/2022 16:35:15 INFO: Removing Wazuh indexer.
24/02/2022 16:35:17 INFO: Wazuh indexer removed.
24/02/2022 16:35:17 INFO: Removing Filebeat.
24/02/2022 16:35:18 INFO: Filebeat removed.
24/02/2022 16:35:18 INFO: Removing Wazuh dashboard.
24/02/2022 16:35:21 INFO: Wazuh dashboard removed.
24/02/2022 16:35:21 INFO: Installation cleaned.
24/02/2022 16:35:21 INFO: --- Configuration files ---
24/02/2022 16:35:21 INFO: Generating configuration files.
24/02/2022 16:35:22 INFO: Created /home/vagrant/repo/wazuh-install-files.tar. Contains Wazuh cluster key, certificates, and passwords necessary for installation.
24/02/2022 16:35:28 INFO: --- Wazuh indexer ---
24/02/2022 16:35:28 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 16:36:33 INFO: Wazuh indexer installation finished.
24/02/2022 16:36:33 INFO: Wazuh indexer post-install configuration finished.
24/02/2022 16:36:33 INFO: Starting service wazuh-indexer.
24/02/2022 16:36:42 INFO: Wazuh-indexer service started.
24/02/2022 16:36:42 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 16:36:47 INFO: Wazuh indexer cluster started.
24/02/2022 16:36:47 INFO: --- Wazuh server ---
24/02/2022 16:36:47 INFO: Starting the Wazuh manager installation.
24/02/2022 16:37:17 INFO: Wazuh manager installation finished.
24/02/2022 16:37:17 INFO: Starting service wazuh-manager.
24/02/2022 16:37:33 INFO: Wazuh-manager service started.
24/02/2022 16:37:33 INFO: Starting Filebeat installation.
24/02/2022 16:37:41 INFO: Filebeat installation finished.
24/02/2022 16:37:42 INFO: Filebeat post-install configuration finished.
24/02/2022 16:37:42 INFO: Starting service filebeat.
24/02/2022 16:37:43 INFO: Filebeat service started.
24/02/2022 16:37:43 INFO: --- Wazuh dashboard ---
24/02/2022 16:37:43 INFO: Starting Wazuh dashboard installation.
24/02/2022 16:38:48 INFO: Wazuh dashboard installation finished.
24/02/2022 16:38:48 INFO: Wazuh dashboard post-install configuration finished.
24/02/2022 16:38:48 INFO: Starting service wazuh-dashboard.
24/02/2022 16:38:48 INFO: Wazuh-dashboard service started.
24/02/2022 16:39:03 INFO: Starting Wazuh dashboard.
24/02/2022 16:39:15 INFO: Wazuh dashboard started.
24/02/2022 16:39:15 INFO: --- Summary ---
24/02/2022 16:39:15 INFO: You can access the web interface https://<wazuh-dashboard-host-ip>.
    User: admin
    Password: GhlWHdFsYJlKT3z9X5zfQbC1EMELYYh7
24/02/2022 16:39:15 INFO: Installation finished.
24/02/2022 16:39:15 INFO: The certificates and passwords used are stored in /home/vagrant/repo/wazuh-install-files.tar.
```

### 2.6: Uninstall

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -u
24/02/2022 16:45:23 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 16:45:23 INFO: Removing Wazuh manager.
24/02/2022 16:45:29 INFO: Wazuh manager removed.
24/02/2022 16:45:29 INFO: Removing Wazuh indexer.
24/02/2022 16:45:32 INFO: Wazuh indexer removed.
24/02/2022 16:45:32 INFO: Removing Filebeat.
24/02/2022 16:45:33 INFO: Filebeat removed.
24/02/2022 16:45:33 INFO: Removing Wazuh dashboard.
24/02/2022 16:45:35 INFO: Wazuh dashboard removed.
```

### 2.7: Uninstall with nothing installed

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -u
24/02/2022 17:17:31 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:17:32 INFO: Wazuh manager components were not found on the system so it was not uninstalled.
24/02/2022 17:17:32 INFO: Filebeat components were not found on the system so it was not uninstalled.
24/02/2022 17:17:32 INFO: Wazuh Indexer components were not found on the system so it was not uninstalled.
24/02/2022 17:17:32 INFO: Wazuh Dashboard components were found on the system so it was not uninstalled.
```

### 2.8: Previous configuration file exists

```
root@ubuntu:/home/vagrant/repo# ls wazuh-install-files.tar 
wazuh-install-files.tar
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a
24/02/2022 17:01:00 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:01:01 INFO: --- Configuration files ---
24/02/2022 17:01:01 INFO: Generating configuration files.
24/02/2022 17:01:01 INFO: Created /home/vagrant/repo/wazuh-install-files.tar. Contains Wazuh cluster key, certificates, and passwords necessary for installation.
24/02/2022 17:01:09 INFO: --- Wazuh indexer ---
24/02/2022 17:01:09 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 17:02:11 INFO: Wazuh indexer installation finished.
24/02/2022 17:02:11 INFO: Wazuh indexer post-install configuration finished.
24/02/2022 17:02:11 INFO: Starting service wazuh-indexer.
24/02/2022 17:02:20 INFO: Wazuh-indexer service started.
24/02/2022 17:02:20 INFO: Initializing Wazuh indexer cluster security settings.
24/02/2022 17:02:25 INFO: Wazuh indexer cluster started.
24/02/2022 17:02:25 INFO: --- Wazuh server ---
24/02/2022 17:02:25 INFO: Starting the Wazuh manager installation.
24/02/2022 17:02:53 INFO: Wazuh manager installation finished.
24/02/2022 17:02:53 INFO: Starting service wazuh-manager.
24/02/2022 17:03:08 INFO: Wazuh-manager service started.
24/02/2022 17:03:08 INFO: Starting Filebeat installation.
24/02/2022 17:03:14 INFO: Filebeat installation finished.
24/02/2022 17:03:15 INFO: Filebeat post-install configuration finished.
24/02/2022 17:03:15 INFO: Starting service filebeat.
24/02/2022 17:03:16 INFO: Filebeat service started.
24/02/2022 17:03:16 INFO: --- Wazuh dashboard ---
24/02/2022 17:03:16 INFO: Starting Wazuh dashboard installation.
24/02/2022 17:04:01 INFO: Wazuh dashboard installation finished.
24/02/2022 17:04:01 INFO: Wazuh dashboard post-install configuration finished.
24/02/2022 17:04:01 INFO: Starting service wazuh-dashboard.
24/02/2022 17:04:02 INFO: Wazuh-dashboard service started.
24/02/2022 17:04:16 INFO: Starting Wazuh dashboard.
24/02/2022 17:04:27 INFO: Wazuh dashboard started.
24/02/2022 17:04:27 INFO: --- Summary ---
24/02/2022 17:04:27 INFO: You can access the web interface https://<wazuh-dashboard-host-ip>.
    User: admin
    Password: X3GAvEgmBi3GEplPMx5uUH1IFiLQM1e6
24/02/2022 17:04:27 INFO: Installation finished.
24/02/2022 17:04:27 INFO: The certificates and passwords used are stored in /home/vagrant/repo/wazuh-install-files.tar.
```

### 2.9: Use of incompatible arguments

```
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -wi indexer
24/02/2022 17:12:14 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:12:15 ERROR: Argument -a|--all-in-one is not compatible with -wi|--wazuh-indexer, -wd|--wazuh-dashboard or -ws|--wazuh-server
root@ubuntu:/home/vagrant/repo# 
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -wi indexer
24/02/2022 17:12:19 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:12:20 ERROR: Argument -a|--all-in-one is not compatible with -wi|--wazuh-indexer, -wd|--wazuh-dashboard or -ws|--wazuh-server
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -ws server
24/02/2022 17:12:27 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:12:27 ERROR: Argument -a|--all-in-one is not compatible with -wi|--wazuh-indexer, -wd|--wazuh-dashboard or -ws|--wazuh-server
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -wd dshboard
24/02/2022 17:12:36 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:12:37 ERROR: Argument -a|--all-in-one is not compatible with -wi|--wazuh-indexer, -wd|--wazuh-dashboard or -ws|--wazuh-server
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -u
24/02/2022 17:22:22 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:22:23 ERROR: It is not possible to uninstall and install in the same operation. If you want to overwrite the components use -o|--overwrite.
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -g
24/02/2022 17:24:07 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:24:08 ERROR: The argument -g|--generate-configurations can't be used with -a|--all-in-one, -o|--overwrite, -s|--start-cluster, -t|--tar, -u|--uninstall, -wd|--wazuh-dashboard, -wi|--wazuh-indexer, or -ws|--wazuh-server
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -F
24/02/2022 17:28:20 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:28:21 ERROR: The -F|--force-dashboard argument needs to be used alongside -wd|--wazuh-dashboard.
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -s
24/02/2022 17:30:56 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.0
24/02/2022 17:30:57 ERROR: The argument -s|--start-cluster can't be used with -a|--all-in-one, -g|--generate-configurations,-o|--overwrite , -u|--uninstall, -wi|--wazuh-indexer, -wd|--wazuh-dashboard, -s|--start-cluster, -ws|--wazuh-server.
root@ubuntu:/home/vagrant/repo# bash wazuh-install.sh -a -h

NAME
        wazuh-install.sh - Install and configure Wazuh central components.

SYNOPSIS
        wazuh-install.sh [OPTIONS] -a | -c | -s | -wi <indexer-node-name> | -wd <dashboards-node-name> | -ws <wazuh-node-name>

DESCRIPTION
        -a,  --all-in-one
                Install and configure Wazuh server, Wazuh indexer, Wazuh dashboard and filebeat

        -c,  --configfile <path-to-config-yml>
                Path to config file. By default: /home/vagrant/repo/config.yml

        -ds,  --disable-spinner
                Disables the spinner indicator.

        -F,  --force-dashboard
                Ignore indexer cluster related errors in Wazuh Dashboard installation

        -g,  --generate-configurations
                Generate /home/vagrant/repo/wazuh-install-files.tar file from /home/vagrant/repo/config.yml containing the files that will be needed for installation. You will need to copy this file to other hosts in distributed deployments.

        -h,  --help
                Display this help and exit.

        -i,  --ignore-hardware-check
                Ignore check for minimum hardware requirements

        -o,  --overwrite
                Overwrites previously installed components. This will erase all the existing configuration and data.

        -s,  --start-cluster
                Starts the indexer cluster.

        -t,  --tar <path-to-certs-tar>
                Path to tar containing certificate files. By default: /home/vagrant/repo/wazuh-install-files.tar

        -u,  --uninstall
                Uninstalls all Wazuh components. This will erase all the existing configuration and data.

        -v,  --verbose
                Shows the complete installation output.

        -V,  --version
                Shows the version of the script and installed packages

        -wd,  --wazuh-dashboards <dashboards-node-name>
                Install and configure Wazuh dashboard.

        -wi,  --wazuh-indexer <indexer-node-name>
                Install and configure Wazuh indexer.

        -ws,  --wazuh-server <wazuh-node-name>
                Install and configure Wazuh server and filebeat.
```

**Comment by alberpilot** [comment]
## Use case 1: Build installer script
Please change the permissions to 500.

**Review by alberpilot — APPROVED** [review]
I approve this PR in order to iterate using this code as a base.

**Pull request merged**